ground/src/main/java/com/google/android/ground/persistence/local/room/entity/LocationOfInterestEntity.java

```diff
@@ -130,9 +132,10 @@ public static LocationOfInterestEntity fromLocationOfInterest(
             .setCreated(AuditInfoEntity.fromObject(locationOfInterest.getCreated()))
             .setLastModified(AuditInfoEntity.fromObject(locationOfInterest.getLastModified()));
     if (locationOfInterest.getType() == LocationOfInterestType.POINT) {
-      entity.setLocation(Coordinates.fromPoint(locationOfInterest.getCoordinatesAsPoint()));
+      entity.setLocation(
+          Coordinates.fromPoint(locationOfInterest.getGeometry().getVertices().get(0)));
     } else {
-      entity.setPolygonVertices(formatVertices(locationOfInterest.getCoordinatesAsPoints()));
+      entity.setPolygonVertices(formatVertices(locationOfInterest.getGeometry().getVertices()));
```

Comment:
This will only work for polygons with no holes. Could you find a TODO + bug to load/save the new model in the local db?

Reply:
#1247